### PR TITLE
Add RotatedBricks

### PR DESCRIPTION
**Pull request by mar-celine**
## Proposed changes

Closes issue #293 

This PR will be rebased on a commit of @fmahebert where the multicube figure domain code is updated to include the code that constructs this figure.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Domain/DomainCreators/RotatedBricks.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/DomainCreators/RotatedBricks.hpp"
+
+#include "DataStructures/Index.hpp"  // for Index
```
Comment (kidder) [inline src/Domain/DomainCreators/RotatedBricks.cpp]: remove comment

File: src/Domain/DomainCreators/RotatedBricks.hpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+/// followed by the rotation U.
+///
+/// - The -x, +y, +z block has the orientation (zeta, -xi, -eta).
+/// It corresponds to the orientation obtained by the rotation R
+/// followed by the rotation U (equivalently, F followed by R).
+///
+/// - The +x, +y, +z block has the orientation (-xi, -eta, -zeta).
+/// It corresponds to the orientation obtained by the rotation F
+/// followed by R followed by U.
+///
```
Comment (kidder) [inline src/Domain/DomainCreators/RotatedBricks.hpp]: We would like to have at least one shared face have all three axes different

File: src/Domain/DomainCreators/RotatedBricks.hpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  struct UpperBound {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {
+        "Sequence of [x,y,z] for upper bounds in the target frame."};
+  };
+
+  struct IsPeriodicIn {
+    using type = std::array<bool, 3>;
+    static constexpr OptionString help = {
+        "Sequence for [x], true if periodic."};
```
Comment (kidder) [inline src/Domain/DomainCreators/RotatedBricks.hpp]: oops

File: src/Domain/DomainCreators/RotatedBricks.hpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  struct InitialRefinement {
+    using type = std::array<size_t, 3>;
+    static constexpr OptionString help = {
+        "Initial refinement level in [x, y, z]."};
+  };
+
+  struct InitialGridPoints {
+    using type = std::array<std::array<size_t, 2>, 3>;
+    static constexpr OptionString help = {
+        "Initial number of grid points in [[x], [y]], [[z]]."};
```
Comment (kidder) [inline src/Domain/DomainCreators/RotatedBricks.hpp]: oops

**Review by kidder — DISMISSED** [review]

File: docs/Images/Tutorials/make_multicube_figures.py [hunk 5]
```diff
@@ -273,8 +273,8 @@ def red_black_blue(mean_xyz):
                        flip_vector=[False, True, True])
     draw_basis_vectors(ax, [0.17, 2.17, 2.3], label_permutation=[2, 0, 1],
                        flip_vector=[False, True, True])
-    draw_basis_vectors(ax, [2.17, 2.17, 2.3], label_permutation=[0, 2, 1],
-                       flip_vector=[True, True, True])
+    draw_basis_vectors(ax, [2.17, 2.17, 2.3], label_permutation=[0, 1, 2],
```
Comment (fmahebert) [inline docs/Images/Tutorials/make_multicube_figures.py]: I would increase the z-position from 2.3 to 2.45, to (visually) separate the basis vectors from the overlapping cube.

Also you now use the default values of `label_permutation` and `flip_vector`, so you can just remove these args as in line 263.

File: src/Domain/DomainCreators/RotatedBricks.hpp [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+///
+/// For reference, this is the notation used:
+///
+///  - U - A clockwise 90 degree rotation about the +z axis (The "up" side)
+///
+///  - R - A clockwise 90 degree rotation about the +x axis (The "right" side)
+///
+///  - F - A clockwise 90 degree rotation about the -y axis (The "front" side)
+///
+///    For reference, D, L, and B ("down", "left", and "back") are the inverse
```
Comment (mar-celine) [inline src/Domain/DomainCreators/RotatedBricks.hpp]: try unindenting this

**Review by nilsdeppe — APPROVED** [review]

**Review by osheamonn — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Travis CI failures are because of timeouts backing up cache in GCC6 builds. Tests pass on both.

**Pull request merged**